packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -163,39 +172,27 @@ class FlatButton extends MaterialButton {
     return RawMaterialButton(
       onPressed: onPressed,
       onHighlightChanged: onHighlightChanged,
-      clipBehavior: clipBehavior ?? Clip.none,
       fillColor: buttonTheme.getFillColor(this),
       textStyle: theme.textTheme.button.copyWith(color: buttonTheme.getTextColor(this)),
+      focusColor: buttonTheme.getFocusColor(this),
+      hoverColor: buttonTheme.getHoverColor(this),
       highlightColor: buttonTheme.getHighlightColor(this),
       splashColor: buttonTheme.getSplashColor(this),
       elevation: buttonTheme.getElevation(this),
+      focusElevation: buttonTheme.getFocusElevation(this),
+      hoverElevation: buttonTheme.getHoverElevation(this),
       highlightElevation: buttonTheme.getHighlightElevation(this),
       disabledElevation: buttonTheme.getDisabledElevation(this),
       padding: buttonTheme.getPadding(this),
       constraints: buttonTheme.getConstraints(this),
       shape: buttonTheme.getShape(this),
-      animationDuration: buttonTheme.getAnimationDuration(this),
+      clipBehavior: clipBehavior ?? Clip.none,
+      focusNode: focusNode,
       materialTapTargetSize: buttonTheme.getMaterialTapTargetSize(this),
+      animationDuration: buttonTheme.getAnimationDuration(this),
       child: child,
     );
   }
-
-  @override
```

Comment:
Yeah, I moved these all up to `MaterialButton`, from all of the `MaterialButton` subclasses, since they all come from there anyhow, and the output was inconsistent between subclasses.  The subclasses should only have things that are additional over `MaterialButton`.